test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1

```diff
@@ -117,6 +117,33 @@ Describe "New-Item" -Tags "CI" {
         $fileInfo.Target | Should -BeNullOrEmpty
         $fileInfo.LinkType | Should -BeExactly "HardLink"
     }
+
+    It "Should create a file at the root of the drive while the current working directory is not the root" {
```

Comment:
The tests are wrong. They pass even without changes in this PR.

To repro this issue using "TestDrive:\", you need to first `Set-Location TestDrive:\`, and then run `New-Item` in the context of the `TestDrive:\`. You cannot just use regular filesystem drive path to repro this.

```
Executing script .\testPR6600.ps1

  Describing New-Item with links
    [+] Should create a file at the root of the drive while the current working directory is not the root 397ms
    [+] Should create a folder at the root of the drive while the current working directory is not the root 80ms
Tests completed in 477ms
Tests Passed: 2, Failed: 0, Skipped: 0, Pending: 0, Inconclusive: 0
```

Reply:
Please use `[Feature]` when add commit to fix the feedback.

Reply:
So should I avoid using $tmpDirectory altogether?

Reply:
Yes, I believe that's how you can repro the issue with `TestDrive:\`. When using `$TestDirve`, you are actually using the real filesystem path, which is not a root path at all.
After you write new tests, please make sure that they fail as expected without your fix, but works with your fix.

Reply:
Thank you, I'll be more explicit about `TestDrive:\` and will check with my new tests. I'll reply here when I confirm.

Reply:
```
Executing script .\test\powershell\Modules\Microsoft.PowerShell.Management\New-Item.Tests.ps1
    Describing New-Item
    [-] Should create a file at the root of the drive while the current working directory is not the root 107ms
      Expected path 'C:\Users\mbobke\AppData\Local\Temp\42bafa09-bdae-4f1d-a18f-33def150c3a3\testfile.txt' to exist, but it did not exist.
      127:             $FullyQualifiedFile | Should -Exist
      at <ScriptBlock>, C:\Users\mbobke\Documents\GitKraken\PowerShell\test\powershell\Modules\Microsoft.PowerShell.Management\New-Item.Tests.ps1: line 127

Confirm
The item at
C:\Users\mbobke\AppData\Local\Temp\42bafa09-bdae-4f1d-a18f-33def150c3a3\newDirectory has
children and the Recurse parameter was not specified. If you continue, all children will be
 removed with the item. Are you sure you want to continue?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): a
    [-] Should create a folder at the root of the drive while the current working directory is not the root 15.96s
      Expected path 'C:\Users\mbobke\AppData\Local\Temp\42bafa09-bdae-4f1d-a18f-33def150c3a3      ewDirectory2' to exist, but it did not exist.
      144:             $FullyQualifiedFolder2 | Should -Exist
      at <ScriptBlock>, C:\Users\mbobke\Documents\GitKraken\PowerShell\test\powershell\Modules\Microsoft.PowerShell.Management\New-Item.Tests.ps1: line 144
```
Confirmed that these tests fail without my changes implemented. Waiting for CI/CD to finish with the changes.

Reply:
Thanks @mcbobke !
It would be easier to review if you push new commit instead of rebasing the existing one, so that we know where we were and only need to look at the new commits.

Also, please add the `[Feature]` tag to your commit message so that CI runs extra tests for the file system code change.

Reply:
I will add the [Feature] tag for you.

Reply:
Understood, I was under the impression that I should try and keep it to a single commit at all times, noted for the future. Sorry about the confusion about the tag, I thought it needed to be in the first line of the body. Thanks for the help!